fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java

```diff
@@ -799,6 +797,11 @@ private void handleRequestDisallowedOnMemento() {
     private FedoraId mintNewPid(final FedoraId fedoraId, final String slug) {
         final String pid;
 
+        if (!isBlank(slug) && slug.contains(" ")) {
+            // Can't use a literal space
+            throw new CannotCreateResourceException("Slug cannot include literal spaces");
```

Comment:
Why are spaces handled differently from any other character that might otherwise be url encoded? For example, is it okay if a slug contains a literal tab character? If so, what is it about spaces that requires special handling? (I'm not saying it doesn't require special handling, just trying to understand.)

Generally speaking, slugs are treated as string literals, yes? So, if I create an object with a slug of `%D0%BE%D0%B1%D1%8A%D0%B5%D0%BA%D1%82` is that the object's literal id or is it `объект`?

Reply:
Sorry I misunderstood and misspoke. My understanding is we allow `-H"Slug: has%20spaces"` but we don't allow `-H:"Slug: has spaces"` because literal spaces are not allowed as part of a valid URI as defined in [RFC-3986](https://tools.ietf.org/html/rfc3986)

Reply:
I think what you are asking is "are we using the URL encoded or decoded form of the path". We are using the decoded form with the exception of spaces. We can move to using the encoded form instead, but currently Jersey decodes on the way in for us. Then we need to work out how to generate a URI without double encoding the path.

Reply:
I don't see where https://tools.ietf.org/html/rfc3986 says you can't have spaces in URIs.

It would seem to me that a slug needs to be treated either as a string literal or as a url encoded value. It's odd to me that only spaces are required to be encoded. If Jersey is treating it as a url encoded value and it is decoding it before injecting the value in FedoraLdp, wouldn't there always be a " " in a slug that contains `+` or `%20`?

Reply:
My reading is the section 2.2 specifies what characters have a reserved purpose in a URI.
> reserved    = gen-delims / sub-delims
>
> gen-delims  = ":" / "/" / "?" / "#" / "[" / "]" / "@"
>
> sub-delims  = "!" / "$" / "&" / "'" / "(" / ")" / "*" / "+" / "," / ";" / "="

and section 2.3 specifies which characters have an unreserved purpose in a URI.
> Characters that are allowed in a URI but do not have a reserved
   purpose are called unreserved.  These include uppercase and lowercase
   letters, decimal digits, hyphen, period, underscore, and tilde.
> 
> unreserved  = ALPHA / DIGIT / "-" / "." / "_" / "~"

Spaces are not defined in either of those sections and so have no purpose in a URI.

Jersey automatically (I think) handles it when you try to add spaces to the URI you are attempting to access, as you [encountered above](https://github.com/fcrepo/fcrepo/pull/1782#issuecomment-713555728). As `Slug`s are passed in as headers which do allow spaces but they are to meant be part of the URI so we enforce this requirement to have the end-user percent encode their spaces.

Reply:
I have reverted this change for a later attempt.